### PR TITLE
Pass `jobId` to the actionsDownloadInfo controller

### PR DESCRIPTION
**Pull request by BALAGA-GAYATRI**
## Context
In order to have the job level telemetry in the github warehouse, `jobId` need to be passed to this table - `hive_hydro.hydro.github_actions_v0_resolve_action_request`
## Implementation
- In this PR `jobId` is being passed as a query parameter to the controller while downloading the actions info.
- This subsequently gets passed to the launch and launch passes this to dotcom's actions resolve API.
- Inside dotcom this `jobId` will be accepted and added to `hive_hydro.hydro.github_actions_v0_resolve_action_request`.
-  We have arrived at the `jobId` (`executionContext.Root.Id`) by matching it with the one from [CustomerIntelligenceHelper.cs](https://github.com/github/actions-dotnet/blob/1fdd638cf6a31a87a19fe6ade72a953ecf5b640e/Actions/Runtime/Sdk/Server/CustomerIntelligenceHelper.cs#L217)

👉[__ADR__](https://github.com/github/c2c-actions/blob/balaga-gayatri/ADR-for-adding-jobId/docs/adrs/3878-identifying-job-runs-from-gh-warehouse-data.md) for detailed implementation. 

**Tracking items** -
- EPIC - [ACE: Migrate to using Adoption data from dashboards instead of manual queries for Azure+GitHub collaboration reviews and reports](https://github.com/github/Actions-India/issues/52)
- Feature - [Deployments/Job Runs cannot be identified from GH warehouse data](https://github.com/github/c2c-actions/issues/3449)

**Other PRs**
- [Actions-dotnet](https://github.com/github/actions-dotnet/pull/10969)
- [Launch](https://github.com/github/launch/pull/5002)
- [Dotcom](https://github.com/github/github/pull/207293)
- [Hydro-schemas](https://github.com/github/hydro-schemas/pull/2390)

### CODE REVIEW

File: src/Sdk/DTGenerated/Generated/TaskHttpClientBase.cs [hunk 1]
```diff
@@ -324,20 +324,22 @@ public TaskHttpClientBase(Uri baseUrl, HttpMessageHandler pipeline, bool dispose
...
             Guid planId,
+            Guid jobId,
             ActionReferenceList actionReferenceList,
             object userState = null,
             CancellationToken cancellationToken = default)
         {
             HttpMethod httpMethod = new HttpMethod("POST");
             Guid locationId = new Guid("27d7f831-88c1-4719-8ca1-6a061dad90eb");
-            object routeValues = new { scopeIdentifier = scopeIdentifier, hubName = hubName, planId = planId };
+            object routeValues = new { scopeIdentifier = scopeIdentifier, hubName = hubName, planId = planId, jobId = jobId };
```
Comment (TingluoHuang) [inline src/Sdk/DTGenerated/Generated/TaskHttpClientBase.cs:342]: will this new route without versioning break compat between the latest runner and old GHES.

Comment (BALAGA-GAYATRI) [reply src/Sdk/DTGenerated/Generated/TaskHttpClientBase.cs:342]: If it is a query parameter change without route value , would it still break the compat and do we need GHES testing in that case?

Comment (TingluoHuang) [reply src/Sdk/DTGenerated/Generated/TaskHttpClientBase.cs:342]: I would suggest you always test GHES with any REST API changes.

File: src/Sdk/DTGenerated/Generated/TaskHttpClientBase.cs [hunk 2]
```diff
@@ -324,27 +324,32 @@ public TaskHttpClientBase(Uri baseUrl, HttpMessageHandler pipeline, bool dispose
...
             Guid planId,
+            Guid jobId,
             ActionReferenceList actionReferenceList,
             object userState = null,
             CancellationToken cancellationToken = default)
         {
             HttpMethod httpMethod = new HttpMethod("POST");
             Guid locationId = new Guid("27d7f831-88c1-4719-8ca1-6a061dad90eb");
-            object routeValues = new { scopeIdentifier = scopeIdentifier, hubName = hubName, planId = planId };
+            object routeValues = new { scopeIdentifier = scopeIdentifier, hubName = hubName, planId = planId};
```
Comment (lokesh755) [inline src/Sdk/DTGenerated/Generated/TaskHttpClientBase.cs:342]:
```suggestion
            object routeValues = new { scopeIdentifier = scopeIdentifier, hubName = hubName, planId = planId };
```

**Review by TingluoHuang — APPROVED** [review]

**Pull request merged**